gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergCatalog.java

```diff
@@ -17,15 +17,19 @@
 
 package org.apache.gobblin.data.management.copy.iceberg;
 
+import java.io.IOException;
 import java.util.Map;
+
 import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.catalog.TableIdentifier;
 
 
 /**
  * Any catalog from which to access {@link IcebergTable}s.
  */
 public interface IcebergCatalog {
-  IcebergTable openTable(String dbName, String tableName);
+  IcebergTable openTable(String dbName, String tableName) throws IOException;
   String getCatalogUri();
   void initialize(Map<String, String> properties, Configuration configuration);
+  boolean tableAlreadyExists(TableIdentifier tableIdentifier);
```

Comment:
super helpful to be able to ask this Q, but why not as a method (e.g. on `IcebergTable`)?  that way someone holding a table can directly inquire about what they have.

...BTW I don't see where it's called.  did you forget to commit that to this PR's branch?

Reply:
That makes sense. I have added that as a pre-check while creating the `IcebergDataset` having both src and dest `IcebergTable`... thus enforcing it to be present always!